### PR TITLE
Initializable

### PR DESCRIPTION
**Pull request by alionako**

### CODE REVIEW

File: Sources/Protocols/Initializable.swift [hunk 1]
```diff
@@ -0,0 +1,39 @@
+//
+//  Copyright (c) 2017 Touch Instinct
```
Comment (ArturAzarau) [inline Sources/Protocols/Initializable.swift]: now it's 2019

File: Sources/Protocols/Initializable.swift [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+//
+
+import Foundation
+
+/// Protocol for initializing an array of instances
+protocol Initializable {
```
Comment (ArturAzarau) [inline Sources/Protocols/Initializable.swift]: If it is a initializable array, maybe it's better to name it `InitializableArray`?

File: Sources/Protocols/Initializable.swift [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    init()
+}
+
+extension Initializable {
+
+    static func instanceArray(count: Int) -> [Self] {
+        var array: [Self] = []
```
Comment (ArturAzarau) [inline Sources/Protocols/Initializable.swift]: Will just `return Array(repeating: Self(), count: count)` work?

File: Sources/Protocols/Initializable.swift [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+import Foundation
+
+/// Protocol for initializing an array of instances
+protocol Initializable {
+    init()
+}
+
+extension Initializable {
+
+    static func instanceArray(count: Int) -> [Self] {
```
Comment (ArturAzarau) [inline Sources/Protocols/Initializable.swift]: proper name for it is `arrayInstance`, not `instanceArray` if it's a noun, and `instantiateArray` if it's a verb

File: Sources/Protocols/Initializable.swift (hunk 1 above)
Comment (alionako) [reply Sources/Protocols/Initializable.swift]: thx!)

File: Sources/Protocols/Initializable.swift [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+        var array: [Self] = []
+        (0..<count).forEach { _ in
+            array.append(Self())
```
Comment (ArturAzarau) [inline Sources/Protocols/Initializable.swift]: will `return (0..<count).map { _ in Self() }` work?

Comment (alionako) [reply Sources/Protocols/Initializable.swift]: no

Comment (alionako) [reply Sources/Protocols/Initializable.swift]: Linter doesn't think that it's a good way to use map (that's why this protocol and pull request appeared, actually)

File: Sources/Protocols/Initializable.swift (hunk 2 above)
Comment (alionako) [reply Sources/Protocols/Initializable.swift]: Discussed protocol naming with @Viveron and @petropavel13, we agreed to call it `Initializable` (which sounds good).

File: Sources/Protocols/Initializable.swift (hunk 4 above)
Comment (Viveron) [reply Sources/Protocols/Initializable.swift]: arrayInstance - нет, так как это массив экземпляров, а не экземпляр массива. instantiateArray - да, согласен

**Pull request merged**